agent/ansible/pbench/agent/roles/pbench_agent_config/tasks/main.yml

```diff
@@ -17,3 +17,8 @@
     dest: "{{ pbench_key_dest }}"
     mode: "0600"
     files: id_rsa
+
+- name: "pbench agent configuration - open firewall ports for tool meister"
```

Comment:
It is not clear to me that we want to unconditionally open firewall ports.  I think we need to offer the role in the documentation, and show how it can be enabled in the playbook (perhaps with a variable?), but this is something the admin needs to be aware of and deliberately perform themselves.

Reply:
What we need is a big warning that the software they're installing *won't work* without these ports open; we can open them, they can open them, or they might as well just quit out of the install right now and forget it.

Implicitly punching holes in their firewall is bad; it definitely needs to be a conscious and tracked decision of the admin. On the other hand, it shouldn't be "oh, the port's not open? no big deal" -- because it is a big deal. I have no idea how we balance this in the ansible; maybe we should give the installer a prompt, and a warning and a paragraph of instruction if they say "no"???

Reply:
Maybe decouple the firewall role from pbench_agent_config completely? It can then be included (or not) in the main playbook, as the user desires. The documentation would include an example playbook including it, but with a comment describing the situation.

The other thing that's missing is the inverse role that undoes everything.